torchdynamo/_guards.cpp

```diff
@@ -52,7 +61,30 @@ class TensorCheck {
       const auto &sizes = v.sizes();
       const auto &strides = v.strides();
       for (size_t i = 0; i < ndim; ++i) {
-        if (sizes_[i] != sizes[i] || strides_[i] != strides[i]) {
+        if (sizes_[i] != sizes[i]) {
+          bool is_dynamic_dim = false;
+          for (int dd = 0; dd < dynamic_dims_.size(); dd++) {
+            auto dim = dynamic_dims_[dd];
+            if (i == dim) {
+              is_dynamic_dim = true;
+              break;
+            }
+          }
+          if (!is_dynamic_dim) {
+            return false;
+          }
+        } else if (strides_[i] != strides[i]) {
+          return false;
```

Comment:
A single dynamic inner dim will cause the stride checks to fail for the outer dims.

Reply:
Sorry, can you elaborate?

Reply:
```
>>> torch.zeros((3,3,3,3)).stride()
(27, 9, 3, 1)
>>> torch.zeros((3,3,3,4)).stride()
(36, 12, 4, 1)
````

I only changed the size of the last dim, but the strides of all the other dims changed.

Reply:
Ah, yea. Let me add a test and make sure I handle this.